ui/address/details/AddressBalance.tsx

```diff
@@ -65,6 +75,66 @@ const AddressBalance = ({ data, isLoading }: Props) => {
     handler: handleNewCoinBalanceMessage,
   });
 
+  const value = data.coin_balance || '0';
+  const exchangeRate = data.exchange_rate;
+  const decimals = String(config.chain.currency.decimals);
+  const accuracyUsd = 2;
+  const accuracy = 8;
+
+  const onGetGasClick = React.useCallback(() => {
+    mixpanel.logEvent(mixpanel.EventTypes.BUTTON_CLICK, { Content: 'Get gas', Source: 'address' });
+  }, []);
+
+  let getGasButton = null;
```

Comment:
Can we move this gas button into a separate component and encapsulate all logic there?